superset/db_engine_specs/mssql.py

```diff
@@ -163,6 +164,33 @@ def extract_error_message(cls, ex: Exception) -> str:
             )
         return f"{cls.engine} error: {cls._extract_error_message(ex)}"
 
+    @classmethod
+    def adjust_query_for_offset(cls, qry: Select) -> Select:
+        """
+        Modify SQLAlchemy query to add ORDER BY when OFFSET is present.
+
+        MSSQL requires an ORDER BY clause when using OFFSET. If no ORDER BY
+        is present but OFFSET is specified, we add a default ORDER BY clause.
+        """
+        # Check if this query has OFFSET but no ORDER BY by inspecting query structure
+        try:
+            # Try to access the internal query structure safely
+            has_offset = getattr(qry, "_offset", None) is not None
+            has_order_by = bool(getattr(qry, "_order_by_clauses", ()))
+
+            if has_offset and not has_order_by:
+                if qry.selected_columns:
+                    first_column = list(qry.selected_columns)[0]
+                    qry = qry.order_by(first_column)
+                else:
```

Comment:
This was AI's idea. If someone feels confident that columns will always be available in this context - seems likely to me - we could remove the if/else and this backup casting of 1.